### PR TITLE
Fix type definition for REPL module of Node.js

### PR DESCRIPTION
**Pull request by ryota-ka**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/repl.html
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -1627,6 +1628,8 @@ declare module "repl" {
         prependOnceListener(event: "reset", listener: (...args: any[]) => void): this;
     }
 
+    export function start(): REPLServer;
+    export function start(prompt: string): REPLServer;
```
Comment (Flarna) [inline types/node/index.d.ts]: You could consider to write this as a single line using `options?: string | ReplOptions`.

**Review by Flarna — COMMENTED** [review]
Adding tests for the new API variants would be nice.

**Comment by RyanCavanaugh** [comment]
@ryota-ka Please fix the failures indicated in the Travis CI log.

**Comment by ryota-ka** [comment]
Thank you for your review.
Fixed up (to be rebased and force pushed), but the build still fails due to #18074.

**Review by Flarna — APPROVED** [review]

**Comment by andy-ms** [comment]
#18074 fixed, please merge latest from master.

**Comment by ryota-ka** [comment]
Rebased and confirmed the build passes.
Please merge if there are no problems.

Thank you for your review and approval!

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**